lib/elements/steel_reinforcement/rebar_hook.rb

```diff
@@ -0,0 +1,31 @@
+require 'db/base'
+require 'elements/steel_reinforcement/base'
+
+module Elements
+  module SteelReinforcement
+    class RebarHook < Base
+      MIN180_HOOK_LENGTH = 60.to_f
+      MIN_HOOK_LENGTH = 75.to_f
```

Comment:
are these constraints coming from design codes? I think we need to have a serious conversation about how to deal with design code constraints. IMO, for now, I think we should keep these classes clean from those values, and in case we need to use them for some calculation we receive them as inputs.

We can later think of how to abstract all the rules or inferences we need to make regarding design codes. What do you think?

One of my biggest concern about this is that one mandatory feature for a program like this is to be able to support multiple design codes, and be prone to changes in future modifications of each code

Reply:
Agree, so in this case, for example, #calculate_length is changed to:

```ruby
def calculate_length
  case angle
  when 90
    12 * diameter
  when 180
    4 * diameter
  else
    6 * diameter
  end
end
```

Reply:
let's remove that method for now, we can work later on how to abstract design codes